### PR TITLE
v2.1 -> v2.2

### PR DESCRIPTION
**Pull request by OlliKiljunen**
# Description of the PR

+ Changes that are needed to make the plug-in compatible for 2021.1. Also, changed version number and such things.

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [x] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt [hunk 1]
```diff
@@ -98,11 +96,6 @@ class PlaceholderTest {
                     }
                     assertEquals("Only one item is selected", selectedItems.size, 1)
                     assertTrue("Selected module is correct", selectedItems[0].contains("Viinaharava", true))
-                    keyboard {
-                        escape()
-                        enterText("'") // a character that won't match anything
-                    }
-                    assertEquals("Nothing should be selected", selectedItems.size, 0)
```
Comment (OlliKiljunen) [inline src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt:105]: In new version of IntelliJ, the selected item stays selected, so this check is no longer needed.

File: src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt [hunk 2]
```diff
@@ -9,9 +9,7 @@ import fi.aalto.cs.apluscourses.e2e.steps.CommonSteps
 import fi.aalto.cs.apluscourses.e2e.utils.StepLoggerInitializer
 import fi.aalto.cs.apluscourses.e2e.utils.getVersion
 import fi.aalto.cs.apluscourses.e2e.utils.uiTest
-import org.junit.Assert.assertEquals
-import org.junit.Assert.assertFalse
-import org.junit.Assert.assertTrue
+import org.junit.Assert.*
```
Comment (nikke234) [inline src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt:12]: Your IDE seems to have some auto-import-optimizing settings on :wink:

**Review by nikke234 — APPROVED** [review]
Great!

**Review by Taikelenn — APPROVED** [review]

**Comment by OlliKiljunen** [comment]
| Manual test            |                                                    |
| ---------------------- | -------------------------------------------------- |
| Date                   | 2021-04-21                                         |
| Operating system       | macOS Catalina                                     |
| IDEA version           | 2021.1                                             |
| Plugin version         | 2.2 (pre-release)                                  |
| Tester                 | @OlliKiljunen                                      |
| Note                   |                                                    |

When the software doesn't comply with the testing manual...

| ...it is marked with... | ...if the reason for the incompliance is that...  |
| ----------------------- | ------------------------------------------------- |
| **FAILED!!!**           | there's a bug in the software.                    |
| **REMARK.**             | the software should (perhaps) be changed.         |
| normal                  | the testing manual should be changed.             |

### Notes

- The REPL part is too long!  Not fun anymore.

- **0.1** The IDEA version should be 2021.1.
- **0.1.B** It happens via File menu as in **0.1.A**.
- **0.2** This option no longer exists.
- **0.3** Restoring settings apparently now remove downloaded plugins.  This step can be removed.
- **1.1** No longer **Configure > Plugins**, just open **Plugins** tab.
- **1.4** The button says **Install**.
- **4.5** Assertion could be more meaningful.
- **4.9** We could test also what happens if cancel is clicked at this phase.
- **4.19** The change of the theme is no longer visible.
- **4.20** Typo in the instruction: "CLick".
- **4.21** Didn't happen because my token was already set.  There should be instructions in the
section **0**, how to clear tokens on each platform.
- **5.17** Clicking the empty background of IDE didn't clear it.  Clicking anything else seems to
work, though.
- **6.13** It's just **Don't ask again**.
- **8.5** Outdated screenshot: icons are different.
- **9.8** **Show History** (sic!) is grayed out.
- **9.9** Omitted.
- **9.12** It should instruct me to close the dialog first.
- **9.14** It should instruct me to close the dialog afterwards.
- **10.1** Should this be only for **Refresh Assignments** now?
- **11.3** I was too slow and missed the hyperlink.
- **11.4** Omitted.
- **12.2** All the options are checked.

**Comment by OlliKiljunen** [comment]
@nikke234's remarks when testing with Ubuntu:
> 4.19 The theme was already dark.
4.21 It first says loading
5.17 Still highlighted
10.2 It would appear without the manual refresh as well because of the post-feedback auto-refresh
11.1 The notification disappears automatically and the hyperlink doesn't work in the event log, so I had to make another submission

**Comment by Taikelenn** [comment]
| Manual test            |                                                    |
| ---------------------- | -------------------------------------------------- |
| Date                   | 2021-04-21                                         |
| Operating system       | Windows 10                                     |
| IDEA version           | 2021.1                                             |
| Plugin version         | 2.2 (pre-release)                                  |
| Tester                 | @Taikelenn                                      |
| Note                   |                                                    |

### Notes

- **0.1** - Now the version is 2021.1
- **1.1** - Remove the "Configure" part, no longer needed
- **3.5** - Scala version is now 2.13.5 but the text shows 2.13.4, giga breaking bug
- **4.10** - "Invalid token" is shown for me because I previously set a wrong token on purpose. Not a big deal, but it would be nice to clean the credentials before starting this step.
- **5.17** - Update to instruct the tester to click on the empty space in the project pane
- **12 (whole section)** - update to cover all filtering options

**Pull request merged**